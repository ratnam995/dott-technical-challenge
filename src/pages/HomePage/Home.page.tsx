import { useNavigate } from 'react-router-dom';

import './home.page.css';

function HomePage() {
  const navigate = useNavigate();

  return (
    <div className='home-wrapper'>
      <h1 className='site-title'>Image Classifier Pro</h1>
      <div className='classifier-list'>
        <button className={'classifier-btn'} onClick={() => navigate("/dog")}>Dogs</button>
      </div>
    </div>
  );
}

export default HomePage;
