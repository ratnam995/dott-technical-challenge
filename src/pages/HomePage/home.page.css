.home-wrapper {
    height: 100vh;
    width: 100vw;
    display: flex;
    flex-direction: column;
}

.site-title {
    text-align: center;
}

.classifier-list {
    display: flex;
    flex-direction: row;
    width: 100%;
    padding: 50px;
}

.classifier-btn {
    width: 300px;
    height: 300px;
    cursor: pointer;
    background-image: url("https://images.dog.ceo/breeds/pug/n02110958_11958.jpg");
    background-repeat: no-repeat;
    background-position: center;
    background-size: cover;
    color: white;
    font-size: 18px;
}
