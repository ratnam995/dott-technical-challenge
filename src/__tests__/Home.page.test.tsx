import renderer from 'react-test-renderer';
import HomePage from '../pages/HomePage/Home.page';

import { BrowserRouter } from 'react-router-dom';

describe("Home page", () => {
    it("renders correctly", () => {
        const tree = renderer
            .create(<BrowserRouter><HomePage/></BrowserRouter>)
            .toJSON();
        expect(tree).toMatchSnapshot();
    });
});