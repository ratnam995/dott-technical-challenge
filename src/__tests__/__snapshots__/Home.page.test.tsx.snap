// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Home page renders correctly 1`] = `
Array [
  <div>
    Image Classifier Pro
  </div>,
  <div>
    <button
      onClick={[Function]}
    >
      Dogs
    </button>
  </div>,
]
`;
